R/import_config.R

```diff
@@ -0,0 +1,173 @@
+#' Get dettl config from path.
+#'
+#' Get the config containing information about location of functions containing
+#' import code.
+#'
+#' Sources any references to functions ready for use. Usea a default
+#' configuration in the result that any fields are missing. This will use the
+#' field name as the expected file name and function anme. Required fields are
+#' extract, transform, test_queries, test and load.
+#'
+#' @param path Path to directory containing config.
+#'
+#' @return The config as a dettl_config object.
+#'
+#' @keywords internal
+#'
+read_config <- function(path) {
+  filename <- file.path(path, "dettl.yml")
+  assert_file_exists(path, name = "Report working directory")
+  assert_file_exists(filename, name = "Dettl configuration")
+  info <- yaml_read(filename)
+
+  ## If certain fields don't exist in the config then add defaults
+  info <- validate_load(info)
+  function_fields <- c(
+    "extract",
+    "transform",
+    "load"
+  )
+  info <- add_missing_function_fields(info, function_fields)
+  required <- c(function_fields, "sources")
+  optional <- c("rewrite_keys")
+  check_fields(info, filename, required, optional)
+  if (info$load$default) {
+    info$rewrite_keys <- ForeignKeyConstraints$new(info$rewrite_keys)
+  }
+  env <- load_sources(info$sources, path)
+  info <- read_function_fields(function_fields, info, env)
+  info$name <- basename(normalizePath(path))
+  info$path <- path
+  class(info) <- "dettl_config"
```

Comment:
You're now in a position where we have two separate objects with this class I believe (the other is created by `dettl_config.R`).  Previously you had `db_config` and `dettl_config` but now they have the same class!  It's up to you how you resolve this, but you might refactor the import to be `dettl_import_config` perhaps?

Reply:
Ah yep, overlooked that. Your suggestion is good, I've fixed that.